### PR TITLE
Refine user profile page

### PR DESCRIPTION
**Pull request by radiofreemattd**
This uses a simple XHR request to return JSON containing the total victories (and some other details) of the user in a modal dialog.

screen shot 2015-12-15 at 5 25 01 pm

### CODE REVIEW

File: app/assets/javascripts/games.js.coffee [hunk 1]
```diff
@@ -8,6 +8,12 @@ $ ->
     channel.bind 'refresh_event', (data)->
       location.reload()
 
+  $('#myModal').on 'show.bs.modal', (e)->
+    userId = $(e.relatedTarget).data('playerid')
+    url = window.location.protocol + '//' + window.location.host + '/users/' + userId
```
Comment (jonmast) [inline app/assets/javascripts/games.js.coffee]: I believe you can leave off the protocol and host as just use a relative url.

File: app/assets/javascripts/games.js.coffee [hunk 2]
```diff
@@ -8,6 +8,12 @@ $ ->
...
+    $.get url, (response)->
+      $('.modal-body').html('<h4>' + response.name + ' has been registered since ' + response.playing_since + ' and has ' + response.total_wins + ' total wins.' + '</h4>')
```
Comment (jonmast) [inline app/assets/javascripts/games.js.coffee]: If the user opens and closes one modal and then opens another they will see the old data until the AJAX loads. Probably not noticeable on locahost but could be somewhat jarring on a slow connection. Maybe we could clear out the modal before we do the ajax call so at least they won't see wrong data?

**Comment by jonmast** [comment]
:+1: Great work. It is stuff like this that adds a polished feel to the app.

File: app/models/user.rb [hunk 3]
```diff
@@ -28,4 +28,12 @@ def games
       player_id: id
     )
   end
+
+  def profile_data
+    { name: username, playing_since: created_at.strftime('%m-%d-%Y'), total_wins: wins }
+  end
+
+  def wins
```
Comment (mscoutermarsh) [inline app/models/user.rb]: i'd call this `wins_count`

File: app/models/user.rb [hunk 4]
```diff
@@ -28,4 +28,12 @@ def games
...
+    Game.where(winner: self).count
```
Comment (mscoutermarsh) [inline app/models/user.rb]: `games.where(winner: self).count`

File: app/models/user.rb [hunk 5]
```diff
@@ -28,4 +28,12 @@ def games
       player_id: id
     )
   end
+
+  def profile_data
+    { name: username, playing_since: created_at.strftime('%m-%d-%Y'), total_wins: wins }
+  end
```
Comment (mscoutermarsh) [inline app/models/user.rb]: love this method. I'd add a test

File: app/controllers/users_controller.rb [hunk 6]
```diff
@@ -6,7 +6,7 @@ def username
 
   def show
     @user = User.find(params[:id])
-    @wins = Game.where(winner: @user).count
+    render json: @user.profile_data
```
Comment (mscoutermarsh) [inline app/controllers/users_controller.rb]: this rocks

**Comment by mscoutermarsh** [comment]
loving the use of rails as a JSON API here. + the nice bit of JS on the frontend.

good stuff. Left a couple comments. Merge whenever you're ready :)

**Pull request merged**